### PR TITLE
fix light pass indices opt

### PR DESCRIPTION
**Pull request by YunHsiao**
Re: https://forum.cocos.org/t/1-2-submodel-lightpassidx/99791

Changelog:
 * light pass indices may very well likely be different among submodels, the assumption doesn't hold.

### CODE REVIEW

File: cocos/core/pipeline/render-additive-light-queue.ts [hunk 1]
```diff
@@ -145,10 +152,7 @@ export class RenderAdditiveLightQueue {
             const ro = this._renderObjects[i];
             const model = ro.model;
             const subModels = model.subModels;
-
-            // this assumes light pass index is the same for all submodels
-            const lightPassIdx = getLightPassIndex(subModels);
-            if (lightPassIdx < 0) continue;
+            if (!getLightPassIndices(subModels, _lightPassIndices)) continue;
```
Comment (pandamicro) [inline cocos/core/pipeline/render-additive-light-queue.ts:155]: This doesn't seem right, (!-1) is false, (!0) is true

Comment (YunHsiao) [reply cocos/core/pipeline/render-additive-light-queue.ts:155]: the return value is a single bool

**Review by pandamicro — APPROVED** [review]

File: cocos/core/pipeline/render-additive-light-queue.ts [hunk 2]
```diff
@@ -170,6 +174,8 @@ export class RenderAdditiveLightQueue {
             if (!_lightIndices.length) continue;
 
             for (let j = 0; j < subModels.length; j++) {
+                const lightPassIdx = _lightPassIndices[j];
+                if (lightPassIdx < 0) continue;
```
Comment (troublemaker52025) [inline cocos/core/pipeline/render-additive-light-queue.ts:178]: 为什么不像以前那样，取得 model 特定位置的 lightPass 进行判断呢？  现在却要把 subModels 里的 lightPass都取出来判断？

Comment (troublemaker52025) [reply cocos/core/pipeline/render-additive-light-queue.ts:178]: @YunHsiao

**Pull request merged**